HLT Prompt DQM for JetMET

physicist87 opened the pull request:
This is for prompt dqm of HLTJetMET.

JetMETMonitor.cc is created for HLTJetMET.

ikrav committed:
Bug fixes, added debug statements

ikrav committed:
Fixed AOD vs miniAOD detection

ikrav committed:
The first version of 80X cut-based photon ID

ikrav committed:
Added the possibility of having different egmPhotonIDSequence for AOD and miniAOD

Conflicts:
	RecoEgamma/PhotonIdentification/python/egmPhotonIDs_cff.py

ikrav committed:
Adjusted python code that deals with different VID sequences for AOD and miniAOD

ikrav committed:
New photon MVA class for 80X releases

ikrav committed:
Another adjustment for the mechanism of having different sequences in VID for AOD and miniAOD

ikrav committed:
Python configuration for the new photon MVA, imcomplete

ikrav committed:
Removed using of shower shapes from PhotonValueMapProducer that was in use until 753. Now shower shapes are taken directly from the photon object

ikrav committed:
More changes for 80X photon MVA python scripts

ikrav committed:
Further modifications to make possible to run different VID sequence for AOD and miniAOD for photons

Conflicts:
	PhysicsTools/SelectorUtils/python/tools/vid_id_tools.py

ikrav committed:
Deleted automatic production of photon MVA IDs that are too old. Left are the most recent 50ns ID (V2p1), 25ns ID (V2p1), and the new 80X ID

ikrav committed:
Fixed a problem with the order of MVA variables

ikrav committed:
Adjusted MD5 sum for the new MVA photon ID

ikrav committed:
Modified photon MVA for 80X: switched to tuning where for EE photon isolation is included in MVA after special scaling

ikrav committed:
Another adjustment of photon MVA configuration: new weights for the endcap, new working point cut values

ikrav committed:
Removed V1 of photon cut-based ID and introduced V2p2 cut-based photon ID which is a bettter tuning for 2016.

ikrav committed:
Changed the input variables rho and worst charged isolation to the form they are derived in the tuning of the latest photon MVA.

ikrav committed:
Updated pointer to indico slides for cut-based 80X photon ID

ikrav committed:
Updated MVA cut values for VID for 80X photon ID after modifying two MVA variables, as committed previously

ikrav committed:
Changes for complience with CMS coding stnadards: position of underscores, import statements, etc

ishvetso committed:
fix to run CITK isolation in VID during miniAOD production

Conflicts:
	PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

ikrav committed:
Replaced import * with explicit imports

Conflicts:
	RecoEgamma/EgammaIsolationAlgos/python/egmPhotonIsolationAOD_cff.py

ishvetso committed:
remove redefinition of pfNoPileUpCandidates

ishvetso committed:
switch to AOD format for switchOnVIDPhotonIdProducer()

ishvetso committed:
fix possible duplications with particleFlowTmpPtrs (added ONLY when the object is not allocated)

ikrav committed:
Removed duplicated import statement

ikrav committed:
A change for PF producer that calls EGM to match the changed naming in EGM python scripts

makortel committed:
Enable pixelPairStep for phase1

makortel committed:
Add BPix1+BPix4 pairs to recover inactive sector in BPix2+BPix3

ikrav committed:
Changes to adjust the new photon ID code to use tasks rather than sequences to match recent changes in related packages in CMSSW

ikrav committed:
A few changes to imrove python coding without change in functioning

makortel committed:
Add maxNVertices to GlobalTrackingRegionWithVerticesProducer

makortel committed:
Sort firstStepPrimaryVertices

makortel committed:
Use only first 5 vertices in PixelPairStep

ikrav committed:
Fixed a bug of using abs that returns an int for the photon MVA

ikrav committed:
Added the latest 80X photon IDs to the default miniAOD production

ikrav committed:
Updated the photon cut-based ID version to the EGM-recommended one for 2016+ data

Commit:
updating FastjetJetProducer

Commit:
adding exceptions

Teemperor committed:
Don't forward declare CaloTower in CaloTowerFwd.h

This header won't compile with just a forward declaration of
CaloTower, because the Ref and RefVector types we declare will check the
ValueTrait of it's containing class, which is in this case CaloTower.
ValueTrait needs to have the definition of CaloTower to work, but
we only have a forward declaration in this header available.

This leads to undefined behavior, so we should replace this forward
declaration with an actual include.

People seem to anyway always include CaloTower too to
avoid this cryptic error, so I don't think this will hit compilation
times that much.

Commit:
adding exceptions

Commit:
adding customize process for FastjetJetProducer

Commit:
fixing custominzeHLTforCMSSW

Commit:
fixing RecoHi

Commit:
fixing puCenters

Sam-Harper committed:
initial version

rovere committed:
Pixel Tracking for 2017 online Beamspot

gartung committed:
fix gcc700 warning: class  has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]

rovere committed:
Add siStrip clusters

rovere committed:
Add siStrip clusters

makortel committed:
Add support for multiple pixelPair seeds, and update time report for the moved modules

Commit:
updating branch

Commit:
removing parameters from VoronoiSubtractor

Commit:
fixing

Sam-Harper committed:
refactor for readability and fill descriptions

mtosi committed:
integrate new HLT objects

mtosi committed:
roll back DQM core config

slehti committed:
HLTTauDQM: changed tau t&p to use 2017 paths

Sam-Harper committed:
updating to use VID and generic trigger event

Sam-Harper committed:
updating to use VID and generic trigger event

slehti committed:
HLTTauDQM: fixed typos in t&p mu+tau50 and ele+tau 2017 paths

kmcdermo committed:
update oot photon sequence names and make changes as suggested

ikrav committed:
Several changes to follow recommendations of code reviewers

ikrav committed:
A fix to make the code compile related to the previous commit

albertdow committed:
adding the latest PFJets/Tau cross cleaning module

makortel committed:
Make BPix1+BPix4 pairs regional

Try to cover only the affected region to reduce fakes and timing.

slehti committed:
HLTTauDQM: fixed two mu+tau paths missing 'eta2p1'

mtosi committed:
fix muon eta binning

Commit:
Add PV information for DIP publication

albertdow committed:
use deltaR2 and consistent member data convention

albertdow committed:
working on a new solution for PFJets Tau overlap

kpedro88 committed:
add fastsim digi aliases for HCAL upgrades

albertdow committed:
changed logic within overlap removal and removed using namespace

albertdow committed:
adding missed std:: for unique_ptr

mseidel42 committed:
Implement comments from @davidlange6

yimuchen committed:
Reverted changes to digis occupancy (implemented in another PR)

yimuchen committed:
Reverted changes to digis occupancy (implemented in another PR)

cschomak committed:
Sign switch back for y-range in GC plots as originally intended and bugfix in labels for difference tables

slehti committed:
HLTTauDQM: a parameter went missing in moving from 2016 to 2017 paths, added back

davidlange6 committed:
trim the size of these luts

rovere committed:
Improve region for pixelTracks

rovere committed:
Center in Z the globalRegion

rovere committed:
Improve logic for output files location

rovere committed:
Improve global tracking region definition

mariadalfonso committed:
add landau for M2 HEP17-shape

Commit:
Correct the output tags

rovere committed:
Improve logic for output files location

makortel committed:
Hack around FastSim

Using unsorted firstStepPrimaryVertices before mixing, and sorted
after mixing is certainly not ideal, but it was the best I could come
up with without addind a circular dependency on 'mix' module via
'ecalRecHit', 'simEcalDigis', 'simEcalTriggerPrimitiveDigis'.

arunhep committed:
MC GTs updated with BPix fixes Fed Cabling,Sim geometry, HCALSiPM Fixes

Commit:
asjust the changes of depth for all PhaseI and PhaseII scenario

kpedro88 committed:
update HBHE raddam parameters

Sam-Harper committed:
allowing VID to gracefully fail if the input collection is invalid

Sam-Harper committed:
allowing VID to gracefully fail if the input collection is invalid

Commit:
Update ieta=16 depths

Sam-Harper committed:
merging in python changes

kmcdermo committed:
include PF cluster isolation into OOT photon sequence

Commit:
Update DQM configuration files for relevant HCAL AlCaReco jobs

mmusich committed:
use native C++ types, use enum instead of string comparisons and use vectors against maps

mseidel42 committed:
More abs->std::abs

mtosi committed:
update

mtosi committed:
add BTV

rmanzoni committed:
vtx smearing parameter for early 2017 MC

mtosi committed:
edm::LogWarning --> LogDebug

yimuchen committed:
Tweaked plot ranges for more descriptive axis title and binning

fabozzi committed:
inject wf without approving

wddgit committed:
Remove service named EnableFloatingPointExceptions

This was discussed and decided at the Core meeting
on 30 May 2017. We do not want to use resources maintaining
this service for several reasons. First of all, as far as we
know no one uses or has used this service and it has been
around for more than 10 years. Its unit test is broken. It
does not work properly in the new multithreaded environment
and would be difficult to fix, the precision control
parts of it are obsolete with modern processors, and it
does not work properly with some architectures.

mariadalfonso committed:
add landau for M3 HEP17-shape

vukasinmilosevic committed:
Adding the template module

vukasinmilosevic committed:
changes vol.2

mmusich committed:
Remove files made useless from AfterAbortGap -> AAG transition; bring back SiStripGainsPCLHarvester

Sam-Harper committed:
reverting changes to VID

Sam-Harper committed:
removing egammaHLT DQM from HI

physicist87 committed:
include PromptDQM for HLTJETMET

physicist87 committed:
update PromptDQM for HLTJETMET

physicist87 committed:
update PromptDQM for HLTJETMET

cmsbuild commented:
A new Pull Request was created by  @physicist87 (Seungkyu Ha) for master.

It involves the following packages:

DQMOffline/Trigger


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

davidlange6 committed:
Merge pull request #18838 from aperloff/fixSpacingJetCorrectorParametersHelper

Fix spacing by replacing all tabs with spaces.

davidlange6 committed:
Merge pull request #18645 from ikrav/egm_id_80X_v3_photons_rebasedTo_CMSSW_9_2_X_2017-05-08-1100

EGM-certified photon IDs for 2016 data: cut based and MVA

fabozzi committed:
cleanup

vukasinmilosevic committed:
Squared the matching dR value, changed names from R to R2

cmsbuild committed:
Merge pull request #18856 from intrepid42/PowhegHooksBB4L_92X

Powheg-bb4l FSR veto

vukasinmilosevic committed:
Added the option of getting the value of matchingR, removed deltaR2

gpetruc committed:
No AssociationVector<bool>

gpetruc committed:
Fix scheduling problems from PATTriggerProducer thinking it may need the TriggerResults of the PAT in which it is running

albertdow committed:
minor changes (parameters,deltaR2,guard string) in both header and source

albertdow committed:
changed deltaR2 function

mtosi commented on DQMOffline/Trigger/plugins/JetMETMonitor.cc:
```diff
@@ -0,0 +1,1056 @@
+  jetmetVsLS_.denominator = nullptr;
+  jetmetPhiME_.numerator   = nullptr;
+  jetmetPhiME_.denominator = nullptr;
+  if (isMetTrig == false) 
```
uhmm, I still do not understand why you cannot use METMonitor for the MET monitor ....

physicist87 replied:
I thought I want to make a code to cover jet trigger and met trigger. 
But It's not efficient way. Your suggestion is better than my code.
So I'm trying to modify this code for only jet. 
And for met trigger, I will use METMonitor code. 
I need to change this code into JetMonitor.cc . Do you agree with me?

mtosi commented on DQMOffline/Trigger/plugins/JetMETMonitor.cc:
```diff
@@ -0,0 +1,1056 @@
+  jetmetPhiME_.denominator = nullptr;
+  if (isMetTrig == false) 
+  {
+     jetmetEtaME_.numerator   = nullptr;
```
what does it mean eta for a MET object ?

physicist87 replied:
Ah ! this is for Jet. 
"isMetTrigger == false " means not Met trigger. only for Jet trigger.
So this jetmetEtaME is for Jet trigger.

mtosi commented on DQMOffline/Trigger/plugins/JetMETMonitor.cc:
```diff
@@ -0,0 +1,1056 @@
+     }
+     if ( int(jets.size()) < njets_ ) return;
+   
+     int ls = iEvent.id().luminosityBlock();
```
this can go above (before the if)

physicist87 replied:
I see. I will move it.

mtosi commented on DQMOffline/Trigger/plugins/JetMETMonitor.cc:
```diff
@@ -0,0 +1,1056 @@
+     } 
+  }// PFJet Trigger 
+  // CaloJet Trigger
+  if (isCaloJetTrig) {
```
couldn't you define a function which handles the histograms
instead of having this copy-n-paste ?

in addition, even if I suggest to develop even further this code allowing for PF specific quantities to be monitored,
as now why don't you make use of the edm::View<reco::Jet> or something like it ?

physicist87 replied:
I will make a function to follow your suggestion.! 
And I will also use edm::Viewreco::Jet. 
Actually, I've never use edm::Viewreco::Jet. Does it have both  Calojet and PFJets ?

mtosi commented on DQMOffline/Trigger/plugins/JetMETMonitor.cc:
```diff
@@ -0,0 +1,1056 @@
+     } 
+
+  }
+  if (isMetTrig) {
```
why don't you use the METMonitor ?!?!??!?!?!!?

physicist87 replied:
Sorry.. I will use the METMonitor. Thanks !!

mtosi commented on DQMOffline/Trigger/plugins/JetMETMonitor.cc:
```diff
@@ -0,0 +1,1056 @@
+  return output;
+}
+/// For Hcal Endcap Plus Area
+bool JetMETMonitor::isEndCapP(double eta){
```
do you really need to have this numbers hardcoded ?
don't we have a map in CMSSW w/ the eta acceptance of our detectors ?

physicist87 replied:
I didn't realize we have a map for detector acceptance.  So I hardcoded ...
Do you have a instruction or twiki to explain  how to use this map?

mtosi commented on DQMOffline/Trigger/python/JetMETMonitor_cff.py:
```diff
@@ -0,0 +1,65 @@
+
+# HLT_PFJet450
+PFJet450_Prommonitoring = hltJetMETmonitoring.clone()
+PFJet450_Prommonitoring.FolderName = cms.string('HLT/JetMET/HLT_PFJet450/')
```
where is HLT_PFJet500 ?

physicist87 replied:
I will update HLT_PFJet500

mtosi commented on DQMOffline/Trigger/python/JetMETMonitor_cff.py:
```diff
@@ -0,0 +1,65 @@
+PFMET170_HBHECleaned_Prommonitoring.ispfjettrg = cms.bool(False)
+PFMET170_HBHECleaned_Prommonitoring.iscalojettrg = cms.bool(False)
+PFMET170_HBHECleaned_Prommonitoring.ismettrg = cms.bool(True)
+PFMET170_HBHECleaned_Prommonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_PFMET170_HBHECleaned_v*")
```
can you point me to the JIRA, please ?
I do not see this trigger path

physicist87 replied:
Actually, this is my first time for Trigger. 
So I don't know how I check the trigger exist in the HLT menu or JIRA. 
If you don't mind, could you please explain how I check the JIRA?

mtosi commented:
I think we are still missing a lot of paths !

cmsbuild committed:
Merge pull request #19062 from fabozzi/from-CMSSW_9_2_X_noapprovewf

Injection of workflows without approving them

cmsbuild committed:
Merge pull request #19074 from gartung/CommonTools-gcc700-dtor-warning-fix

CommonTools/RecoUtils : fix gcc700 warning: class  has virtual functions and accessible non-virtual destructor [-Wnon-virtual-dtor]

Commit:
Update the MinBias trigger path

kmcdermo committed:
include ecal and hcal interesting rechits from pf cluster isolation

davidlange6 committed:
Merge pull request #19036 from bsunanda/Run2-hcx128

Run2-hcx128: adjust the depth assignments in HB for all PhaseI and PhaseII scenario

davidlange6 committed:
Merge pull request #19039 from kpedro88/RadDam-7

update HBHE raddam parameters

davidlange6 committed:
Merge pull request #18866 from igv4321/sipm-nonlinearity-fix

Switching to full pulse nonlinearity for SiPMs

davidlange6 committed:
Merge pull request #19050 from rovere/updateOnlineBSWithPVInformationMaster

Update online bs with pv information master

physicist87 commented:
I misunderstood the list of Jet trigger which we want to monitor. 
I will update other triggers. 
Thank you for your suggestions and comments.

mtosi commented:
please,
the easiest thing is to check w/ JME group
i'm using my phone now
and i'll be offline the whole day :(

  mia

> On 5 Jun 2017, at 03:06, Seungkyu Ha <notifications@github.com> wrote:
> 
> @physicist87 commented on this pull request.
> 
> In DQMOffline/Trigger/python/JetMETMonitor_cff.py:
> 
> > +CaloJet500_NoJetID_Prommonitoring = hltJetMETmonitoring.clone()
> +CaloJet500_NoJetID_Prommonitoring.FolderName = cms.string('HLT/JetMET/HLT_CaloJet500_NoJetID/')
> +CaloJet500_NoJetID_Prommonitoring.ptcut = cms.double(500)
> +CaloJet500_NoJetID_Prommonitoring.ispfjettrg = cms.bool(False)
> +CaloJet500_NoJetID_Prommonitoring.iscalojettrg = cms.bool(True)
> +CaloJet500_NoJetID_Prommonitoring.ismettrg = cms.bool(False)
> +CaloJet500_NoJetID_Prommonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_CaloJet500_NoJetID_v*")
> +
> +# HLT_PFMET170_HBHECleaned
> +PFMET170_HBHECleaned_Prommonitoring = hltJetMETmonitoring.clone()
> +PFMET170_HBHECleaned_Prommonitoring.FolderName = cms.string('HLT/JetMET/HLT_PFMET170_HBHECleaned/')
> +PFMET170_HBHECleaned_Prommonitoring.ptcut = cms.double(170)
> +PFMET170_HBHECleaned_Prommonitoring.ispfjettrg = cms.bool(False)
> +PFMET170_HBHECleaned_Prommonitoring.iscalojettrg = cms.bool(False)
> +PFMET170_HBHECleaned_Prommonitoring.ismettrg = cms.bool(True)
> +PFMET170_HBHECleaned_Prommonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_PFMET170_HBHECleaned_v*")
> Actually, this is my first time for Trigger.
> So I don't know how I check the trigger exist in the HLT menu or JIRA.
> If you don't mind, could you please explain how I check the JIRA?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

mtosi commented:
please, check w/ HCAL DPG

> On 5 Jun 2017, at 03:01, Seungkyu Ha <notifications@github.com> wrote:
> 
> @physicist87 commented on this pull request.
> 
> In DQMOffline/Trigger/plugins/JetMETMonitor.cc:
> 
> > +}
> +
> +bool JetMETMonitor::isBarrel(double eta){
> +  bool output = false;
> +  if (fabs(eta)<=1.3) output=true;
> +  return output;
> +}
> +
> +//------------------------------------------------------------------------//
> +bool JetMETMonitor::isEndCapM(double eta){
> +  bool output = false;
> +  if (fabs(eta) <= 3.0 && fabs(eta) > 1.3 && (eta/fabs(eta) < 0) ) output=true;
> +  return output;
> +}
> +/// For Hcal Endcap Plus Area
> +bool JetMETMonitor::isEndCapP(double eta){
> I didn't realize we have a map for detector acceptance. So I hardcoded ...
> Do you have a instruction or twiki to explain how to use this map?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

mtosi commented:
if you use reco::Jet it should work for both
but i think you want to monitor even calo/pf specific quantities
=> you probably want to leave it separately
or having a common part and the have 2 somatic functions

> On 5 Jun 2017, at 02:59, Seungkyu Ha <notifications@github.com> wrote:
> 
> @physicist87 commented on this pull request.
> 
> In DQMOffline/Trigger/plugins/JetMETMonitor.cc:
> 
> > +        jetmetHEM17_EtaME_.numerator -> Fill(jeteta_);
> +        jetmetHEM17_VsLS_.numerator -> Fill(ls, jetpt_);
> +        jetmetHEM17_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
> +     }
> +     if (isHEP18( jeteta_, jetphi_ ) )
> +     {
> +        jetmetHEP18_ME_.numerator -> Fill(jetpt_);
> +        jetmetHEP18_ME_variableBinning_.numerator -> Fill(jetpt_);
> +        jetmetHEP18_PhiME_.numerator -> Fill(jetphi_);
> +        jetmetHEP18_EtaME_.numerator -> Fill(jeteta_);
> +        jetmetHEP18_VsLS_.numerator -> Fill(ls, jetpt_);
> +        jetmetHEP18_EtaVsPhi_.numerator -> Fill(jeteta_,jetphi_);
> +     } 
> +  }// PFJet Trigger 
> +  // CaloJet Trigger
> +  if (isCaloJetTrig) {
> I will make a function to follow your suggestion.!
> And I will also use edm::Viewreco::Jet.
> Actually, I've never use edm::Viewreco::Jet. Does it have both Calojet and PFJets ?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

fabozzi committed:
updating 2017 gensim recycling to 9_2_1

fabozzi committed:
fixing statistics of JpsiMuMu_Pt-8 2017 relval

ianna committed:
Move DD algorith plugins in plugin directory

ianna committed:
moved headers

perrotta committed:
restore one line that got lost while rebasing

ianna committed:
Merge Base to Core

perrotta committed:
Factorize fillDescriptions

perrotta committed:
fix typo

cmsbuild committed:
Merge pull request #19092 from ianna/dd-algorithm-move

Move DD Algorithm Plugins in a Plugin Directory

perrotta committed:
remove also sumRecHits

kpedro88 committed:
add forward declarations of template specializations

perrotta committed:
Revert "remove also sumRecHits"

This reverts commit 57a4259a134ce05121cf920f07a48817a663d709.

perrotta committed:
Use the correct PR nr in the HLT customization

Commit:
Enabling depth studies in Ecal crystal

mariadalfonso committed:
fix array

Commit:
extending dumping precision for Gains

Teemperor committed:
Renamed CaloTowerFwd.h to CaloTowerDefs.h

Because CaloTowerFwd.h is not really forward declaring anything, we
rename it to CaloTowerDefs.h to better represent it's purpose.

fioriNTU committed:
fix some ranges/binning of plots

davidlange6 committed:
Merge pull request #19080 from gpetruc/no_assovec_bool_to_valuemap_cmssw921

PhysicsTools fixes for Muon Tag & Probe in 92X

kmcdermo committed:
Revert "include ecal and hcal interesting rechits from pf cluster isolation"

This reverts commit 3fb364abebf7283a333cad3bdd703688ba9a5399.

mmusich committed:
fix CalibTracker.Configuration.Common.PoolDBEEsource_cfi to remove disclaimer warning

perrotta committed:
Style fixes, from DavidL

Teemperor committed:
Add vector include to EventVariables.h

The v_sector member is using std::vector, so we need the definition
in this header.

ianna committed:
Merge DD ExprAlgo to Core

Teemperor committed:
Add include for Trajectory.h to TrajGsfTrackAssociation.h

We use the Trajectory class in this header, but we didn't include
the relevant header, which prevents this header from compiling.

Teemperor committed:
Add missing include to ostream to Track/VertexCategories.h

At the end of these files we provide a printing function for
std::ostream, but we also need to include the ostream header before
that.

Teemperor committed:
Add missing includes to CategoryCriteria.h

We use Event, EventSetup and ParameterSet in this header, but we
didn't include the necessary headers. This patch adds the missing
includes.

mariadalfonso committed:
activate the new M3 landau shape only when the GT use the new shape

mtosi commented:
@physicist87 how is it going ?

fioriNTU committed:
adjusting also residuals plots

Commit:
Correct the python script for HCAL pedestal to make it work for different scenario

DryRun committed:
Propagate EDAnalyzer to EDProducer to DQMEDHarver changes to HcalOnlineHarvesting

physicist87 commented:
At first, I removed Met Trigger information in this code to use METMonitor code.
And I changed the name of JetMETMonitor.cc to JetMonitor.cc (only for JetTrigger).
And I'm trying to make a function to fill the histograms for both of pfJet and calojet.

And I'm trying to update other trigger path you said to me.

mtosi commented:
thanks


> On 6 Jun 2017, at 16:35, Seungkyu Ha <notifications@github.com> wrote:
> 
> At first, I removed Met Trigger information in this code to use METMonitor code.
> And I changed the name of JetMETMonitor.cc to JetMonitor.cc (only for JetTrigger).
> And I'm trying to make a function to fill the histograms for both of pfJet and calojet.
> 
> And I'm trying to update other trigger path you said to me.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

perrotta committed:
Merged refs/pull/19098/head from repository cms-sw

davidlange6 committed:
Merge pull request #19077 from MilanoBicocca-pix/master

Vertex smearing parameter for early 2017 MC

davidlange6 committed:
Merge pull request #19054 from bsunanda/Run2-hcx131

Run2-hcx131 Update DQM configuration files for relevant HCAL AlCaReco jobs

cmsbuild committed:
Merge pull request #18967 from kpedro88/Phase1-HF12

Make FastSim work with QIE10/11

davidlange6 committed:
Merge pull request #19031 from davidlange6/calibluts

trim the size of these luts

cmsbuild committed:
Merge pull request #18965 from stahlleiton/OnlineDQM_2DChi2QT_92X

OnlineDQM: Add Chi2 2D Quality Test

perrotta committed:
Do not use addDefault if you want autogenerated cfi's to be parsed for HLT

cmsbuild committed:
Merge pull request #18963 from mmusich/92X_PVValidationUpdates

[92X] PrimaryVertexValidation Updates

cmsbuild committed:
Merge pull request #19114 from DryRun/CMSSW_9_2_1_hcaldqm-fix-online-harvesting

Propagate EDAnalyzer to EDProducer to DQMEDHarvester changes to HcalOnli…

davidlange6 committed:
Merge pull request #19065 from wddgit/removeEnableFloatingPointExceptionsService

Remove service named EnableFloatingPointExceptions

davidlange6 committed:
Merge pull request #19033 from bsunanda/Run2-hcx127

Run2-hcx127 Correct the output cff sepecification for 2 minimum bias paths

kmcdermo committed:
simple clones for interesting ES and HCAL recHit producers

albertdow committed:
moving Rsquared to constructor and removing unnecessary includes

slava77devel committed:
initialize default wrap to false

perrotta committed:
restore @module_label, but remove it from fillDescriptions

cmsbuild committed:
Merge pull request #19116 from mmusich/92X_fixCalibTrackerCommonWarning

Remove warnings from CalibTracker ESSource provider

cmsbuild committed:
Merge pull request #19107 from Teemperor/FixMissingIncludeEventVariables.h

Add vector include to EventVariables.h

cmsbuild committed:
Merge pull request #19099 from bsunanda/Run2-sim08

Run2-sim08 Enabling depth studies in Ecal crystal

cmsbuild committed:
Merge pull request #19061 from enochnotsocool/FEDTrendFix

FED Digis trend plot fix

Teemperor committed:
Add include to MessageLogger.h to EcalEBTrigPrimTestAlgo.h

We use edm::LogError here, so we also need to include the header
that is providing this logging mechanism.

Teemperor committed:
Add missing includes to TTHelpers.h

We use both TransientTrack and TransientTrackBuilder in this header,
so we also need to include the header for those classes. The other
definitions needed to compile this header (e.g. TrackRef) are also
provided by this header, so this should compile now.

Teemperor committed:
Add missing include for BaseDeDxEstimator.h to RecoTracker/DeDx headers.

Those headers all have classes who inherit from BaseDeDxEstimator, so
they also need to include the appropriate header.

cmsbuild committed:
Merge pull request #19103 from fioriNTU/FixRangesPhase192X

fix some ranges/binning of plots in Phase 1 DQM

davidlange6 committed:
Merged refs/pull/19038/head from repository cms-sw

davidlange6 committed:
Merged refs/pull/19091/head from repository cms-sw

vukasinmilosevic committed:
Changes added. Also added condition (myJet.pt()>pt2Min) in the matching part.

vukasinmilosevic committed:
added changes vol.2

cmsbuild committed:
Merge pull request #19120 from slava77/CMSSW_9_2_1/sign921/vgCleanup-mix

initialize default wrap to false in MixingModule Adjuster

davidlange6 committed:
Merge pull request #19106 from ianna/dd-expralgo-cleanup

DD ExprAlgo Merge with Core

mtosi committed:
update FastTimerServiceClient

davidlange6 committed:
Merge pull request #18805 from Teemperor/FixCaloTowerFwd.h

Don't forward declare CaloTower in CaloTowerFwd.h

vukasinmilosevic committed:
Changes vol.3

mtosi committed:
update directory for EndPath

ianna committed:
Migrate to multithreaded analyzers

mariadalfonso committed:
document the 205 shape

mtosi committed:
add tracking monitoring to online beam spot

albertdow committed:
removing initialisation of mindR_ and using only matchingR2_ instead

cmsbuild committed:
Merge pull request #19111 from Teemperor/FixIncludesCategoryCriteria.h

Add missing includes to CategoryCriteria.h

davidlange6 committed:
Merge pull request #19046 from Sam-Harper/GenericTPDQM920

E/gamma T&P Offline DQM : 92X + VID + GenericTrigEventFlag Changes

slehti committed:
HLTTauDQM: changed tagAndProbePaths to be a local parameter

Commit:
select PCC object only from Random producer

Commit:
select PCC object only from ZeroBias  producer

Commit:
no blanks

cmsbuild committed:
Merge pull request #18997 from albertdow/FinalCrossCleaningModule

adding a PFJets-Tau overlap removal module for HLT

cmsbuild committed:
Merge pull request #19130 from ianna/geom-mt-analyzers

Use Thread-safe Analyzers

cmsbuild committed:
Merge pull request #19128 from mtosi/fixFastTimerServiceClient

update FastTimerServiceClient

cmsbuild committed:
Merge pull request #19132 from mtosi/trackingDQM4onlineBS_92x

add tracking monitoring to online beam spot

davidlange6 committed:
Merge pull request #19070 from mariadalfonso/siPMdataM2Shape_92X

HEP17: pulse shape update (M2,M3,M0)

cmsbuild committed:
Merge pull request #19075 from mmusich/92X_remove_unused_configuration

Remove files made useless from AfterAbortGap -> AAG transition

rmanzoni committed:
update Z because of cogging adjustment by LHC

davidlange6 committed:
Merge pull request #19127 from davidlange6/newPixGeom

New pixel geometry in runTheMatrix tests

davidlange6 committed:
Merge pull request #18172 from slehti/HLTTauValidationUpdate_CMSSW91X_April2017

Hlt tau validation update cmssw91 x april2017

davidlange6 committed:
Merge pull request #19109 from Teemperor/VertexCategories.h

Add missing include to ostream to Track/VertexCategories.h

davidlange6 committed:
Merge pull request #19137 from MilanoBicocca-pix/master

update Z because of cogging adjustment by LHC

vukasinmilosevic committed:
Merge conflict resolved

cmsbuild committed:
Merge pull request #19098 from perrotta/CMSSW_9_2_X_FastjetJetProducerUpdateWithFactorizedFilldescriptions

Updating FastjetJetProducer and related producers with (factorized) fillDescriptions

Commit:
Commenting unused variables

fwyzard commented:
tracked by #19142

vukasinmilosevic committed:
Removing .Vect()

cmsbuild committed:
Merge pull request #19136 from franzoni/selective-output-PCC-ZeroBiasVSRandom-from-CMSSW_9_2_1

Pixel Cluster Counting ALCARECO: separate content for random and zerobias

davidlange6 committed:
Merge pull request #19112 from bsunanda/Run2-hcx132

Run2-hcx132 Correct the python script for HCAL pedestal to make it work for different scenario

cmsbuild committed:
Merge pull request #19101 from abdoulline/fix_dump_Gains

Minor tech.fix: extending dumping precision for Gains

cmsbuild committed:
Merge pull request #19096 from kpedro88/FixASCIIIO

Fix HcalDbASCIIIO template specializations

cmsbuild committed:
Merge pull request #18904 from ghellwig/fix-empty-surface-handling_92X

[92X] Apply surface parameters only if selected for alignment.

gpasztor commented:
Is there a progress for 
HLT_PFJet140 / 200 / 260 / 320 / 400_v*
and all of the AK8 version of the PFJet triggers so that they can enter the menu when we restart data taking next week?
Thanks, Gabi

cmsbuild committed:
Merge pull request #18879 from mtosi/onlineDQM4HLT

new HLT objects monitor code

cmsbuild committed:
Merge pull request #19123 from Teemperor/MissingIncludesRecoTracerDeDx

Add missing include for BaseDeDxEstimator.h to RecoTracker/DeDx headers.

cmsbuild committed:
Merge pull request #19122 from Teemperor/FixIncludesTTHelpers.h

Add missing includes to TTHelpers.h

davidlange6 committed:
Merge pull request #19121 from Teemperor/IncludeMessageLoggerInEcalEBTrigPrimTestAlgo.h

Add include to MessageLogger.h to EcalEBTrigPrimTestAlgo.h

cmsbuild committed:
Merge pull request #19108 from Teemperor/FixMissingIncludesTrajGsfTrackAssociation.h

Add include for Trajectory.h to TrajGsfTrackAssociation.h

davidlange6 committed:
Merge pull request #19143 from mrodozov/fix-unused-vars-warnings

Commenting unused variables

davidlange6 committed:
Merge pull request #19051 from makortel/phase1PixelPair

Mitigation for inactive pixel regions for phase1 tracking

cmsbuild committed:
Merge pull request #19030 from cschomak/GeometryComparison_9_2_0_Fixes

Small fixes in GC tool

cmsbuild committed:
Merge pull request #18844 from kmcdermo/oot-photons-aod

Out-of-time photons in RECO/AOD

cmsbuild committed:
Merge pull request #19068 from vukasinmilosevic/master_L1TJetsMatching

L1 pf jets matching - Adding the template module

physicist87 commented:
Hello Gabi and all,
We are updating these trigger list. 
HLT_PFJet140_v*
HLT_PFJet200_v*
HLT_PFJet260_v*
HLT_PFJet320_v*
HLT_PFJet400_v*
HLT_AK8PFJet40_v*
HLT_AK8PFJet60_v*
HLT_AK8PFJet80_v*
HLT_AK8PFJet140_v*
HLT_AK8PFJet200_v*
HLT_AK8PFJet260_v*
HLT_AK8PFJet320_v*
HLT_AK8PFJet400_v*
HLT_AK8PFJet450_v*
HLT_AK8PFJet500_v*
HLT_PFJetFwd40_v9
HLT_PFJetFwd60_v9
HLT_PFJetFwd80_v8
HLT_PFJetFwd140_v8
HLT_PFJetFwd200_v8
HLT_PFJetFwd260_v9
HLT_PFJetFwd320_v9
HLT_PFJetFwd400_v9
HLT_PFJetFwd450_v9
HLT_PFJetFwd500_v9
HLT_AK8PFJetFwd40_v*
HLT_AK8PFJetFwd60_v4
HLT_AK8PFJetFwd80_v4
HLT_AK8PFJetFwd140_v4
HLT_AK8PFJetFwd200_v4
HLT_AK8PFJetFwd260_v*
HLT_AK8PFJetFwd320_v*
HLT_AK8PFJetFwd400_v*
HLT_AK8PFJetFwd450_v*
HLT_AK8PFJetFwd500_v*
And we are selecting  proper pt binning for each trigger path. 

Thanks,
Seungkyu

fwyzard commented:
can you fix the merge conflicts ?
you probably need to rebase this PR on CMSSW 9.2.2 .

gpasztor commented:
Thanks Seungkyu, could you please add also PFJet550 and AK8PFJet550 to your list?
Many thanks, Gabi

sehwook commented:
Dear Gabi,
Right now, Seungkyu is updating pull request which should be done as soon as possible.
Let him update the PR first.
Sehwook

physicist87 committed:
Update Prompt DQM for HLTJet

physicist87 committed:
update PromptDQM for HLTJET

physicist87 committed:
update PromptDQM for HLTJET

physicist87 committed:
update PromptDQM for HLTJET

physicist87 committed:
update PromptDQM for HLTJET

physicist87 committed:
update PromptDQM for HLTJET

physicist87 committed:
update PromptDQM for HLTJET

cmsbuild commented:
Pull request #19076 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

physicist87 committed:
Remove DQMOffline/Trigger/plugins/JetMETMonitor.cc

physicist87 committed:
Remove DQMOffline/Trigger/plugins/JetMETMonitor.h

physicist87 committed:
Remove Trigger/python/JetMETMonitor_cff.py

physicist87 committed:
Remove Trigger/python/JetMETMonitor_cfi.py

cmsbuild commented:
Pull request #19076 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

physicist87 committed:
Merged hltjetmet-promptqdm-updateV1 from repository physicist87

Conflicts:
	DQMOffline/Trigger/python/DQMOffline_Trigger_cff.py

cmsbuild commented:
Pull request #19076 was updated. @ghellwig, @lveldere, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @smuzaffar, @Dr15Jones, @kkousour, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @ssekmen, @arunhep, @perrozzi, @thuer, @dmitrijus, @alja, @franzoni can you please check and sign again.

kpedro88 commented:
-1
bad merge
please follow http://cms-sw.github.io/tutorial-resolve-conflicts.html

slava77 commented:
-1

bad merge

ianna commented:
-1

davidlange6 commented:
please close your obsolete pull requests. Thanks
